src/platform/iotjs_module_uart-linux-general.inl.h

```diff
@@ -135,6 +135,14 @@ void OpenWorkerUart(uv_work_t* work_req) {
 }
 
 
+void UartClose() {
+  uv_close((uv_handle_t*)&poll_handle, NULL);
+  if (fd > 0) {
```

Comment:
If you see line 31,  `fd` is just declared with no initialization. So It is likely to be set to 0. `fd` can be 0 even though the file descriptor is invalid. If the code should be changed according to your suggestion, `fd` should be initialized to non-positive number on line 31.

Apart from your suggestion, I found a bug here. Line 139 should be executed after line 140's condition. I'll fix the bug immediately.